### PR TITLE
Card 427 store fired rules

### PR DESCRIPTION
**Pull request by wezell**
This pull request does 2 things. 
1. It puts a List of the rules fired as a request attribute which can be examined by a web developer looking to debug their rules: 

```
#foreach($rule in $request.getAttribute("dotRulesFired"))
$rule<br>
#end
```
1. It allows all rules to be ignored, by passing dotRules=skip as a url param.  

I am not in love with the WebKeys and am open to suggestions.   please review, modify as needed and merge

### CODE REVIEW

File: src/com/dotmarketing/portlets/rules/business/RulesEngine.java [hunk 1]
```diff
@@ -3,36 +3,42 @@
...
         }
-
-        Host host;
-
-        try {
-            host =  WebAPILocator.getHostWebAPI().getCurrentHost(req);
-        } catch (Exception e) {
-            Logger.error(RulesEngine.class, "Unable to retrieve current request host for URI ", e);
-            return;
+        if(req !=null ){
```
Comment (ggranum) [inline src/com/dotmarketing/portlets/rules/business/RulesEngine.java]: can request ever be null? I'd consider that an exception myself.

Comment (wezell) [inline src/com/dotmarketing/portlets/rules/business/RulesEngine.java]: It probably should be - I was just being wary of all our page generation type, like the timemachine or any unit tests.  We could throw there, np

File: src/com/dotmarketing/portlets/rules/business/RulesEngine.java [hunk 2]
```diff
@@ -44,13 +50,32 @@ public static void fireRules(HttpServletRequest req, HttpServletResponse res, Ru
...
         try {
 
-            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(host.getIdentifier(), systemUser, false, fireOn);
-
+            Set<Rule> rules = APILocator.getRulesAPI().getRulesByParentFireOn(parent.getIdentifier(), systemUser, false, fireOn);
+			List<Rule> firedRules = (List<Rule>) req.getAttribute(WebKeys.RULES_ENGINE_FIRE_LIST);
+        	if(firedRules ==null){
+        		firedRules = new ArrayList<Rule>();
+        		req.setAttribute(WebKeys.RULES_ENGINE_FIRE_LIST, firedRules);
+        	}
```
Comment (brentgriffin) [inline src/com/dotmarketing/portlets/rules/business/RulesEngine.java]: Need to keep in mind that not all rules fire on a per request/page basis - believe that some rules firing status will need to be kept in the session - i.e. for "Once per visit" and "Once per visitor" firing frequency will cause rules to fire that may affect all of the requests in this session (by altering state) but will only fire on the first request of the users session

Comment (wezell) [inline src/com/dotmarketing/portlets/rules/business/RulesEngine.java]: right, I thought that too.  I just am wary of adding more to the session.  I was thinking we could use a flash session (which I have always thought we needed).  Maybe phase 2?

**Comment by brentgriffin** [comment]
Functionality merged here:  https://github.com/dotCMS/core/pull/8632

**Pull request closed without merging**